### PR TITLE
minor refactor

### PR DESCRIPTION
**Pull request by zvolchak**

### CODE REVIEW

File: utils/customize_node.py [hunk 1]
```diff
@@ -188,9 +201,11 @@ def set_apt_proxy(args):
     # insufficient.  Either way, get the proxies from os.environ.
     path = args.new_fs_dir + '/etc/apt/apt.conf.d'
     file_utils.make_dir(path)
-    path += '/00FAMproxy.conf'  # !!! FIXME: no hardcoded custom file names !!!
+    path += '/00TMMS.conf'
```
[the remark this answers is not in the document]
Comment (zvolchak) [reply utils/customize_node.py]: No. You created this code... :)

File: utils/customize_node.py [hunk 2]
```diff
@@ -48,16 +48,27 @@ def extract_bootfiles(args, keep_kernel=False):
...
     vmlinuz = glob.glob('%s/vmlinuz*' % (boot_dir))      # Move and keep
     initrd = glob.glob('%s/initrd.img*' % (boot_dir))    # Move and ignore
-    #also extract config-* and System.map* files installed with kernel.
+
+    is_force_keep_kernel = False
+    # 'manifest' param may not exist when building golden image.
+    if hasattr(args, 'manifest'):
+        #override 'keep_kernel' if user used 'keep_kernel' in the manifest
+        is_force_keep_kernel = getattr(args.manifest, 'keep_kernel', False)
+        keep_kernel = is_force_keep_kernel
```
[the remark this answers is not in the document]
Comment (zvolchak) [reply utils/customize_node.py]: "is_force_keep_kernel" overrides "keep_kernel" state. You right, that can be replaced, but in a different spot, actually.

File: blueprints/30-nodes/blueprint.py [hunk 3]
```diff
@@ -251,15 +252,24 @@ def delete_node_binding(nodespec):
 
     try:
         node_image_dir = node_coord2image_dir(node_coord)
-        for node_file in glob.glob(node_image_dir + '/*'):
-            os.remove(node_file)
+        node_name = os.path.basename(node_image_dir)
+        node_build_dir = BP.config['FILESYSTEM_IMAGES'] + '/' + node_name
```
[the remark this answers is not in the document]
Comment (zvolchak) [reply blueprints/30-nodes/blueprint.py]: "node_coord2image_dir" is for tmms/tftp/ location. Here, I need tmms/sys-images/ build directory.

File: utils/customize_node.py [hunk 4]
```diff
@@ -1109,7 +1125,7 @@ def execute(args):
         args.debug = False
     if getattr(args, 'verbose', None) is None:
         args.verbose = False
-    if getattr(args, 'is_golden', None) is None:    # setup_golden.py
+    if getattr(args, 'is_golden', None) is None:    # set in setup_golden.py
```
[the remark this answers is not in the document]
Comment (zvolchak) [reply utils/customize_node.py]: no, this is the only one (unless I'm missing something?).

File: utils/customize_node.py [hunk 5]
```diff
@@ -1162,13 +1178,12 @@ def execute(args):
         args.new_fs_dir = core_utils.untar(args.build_dir + '/untar/', args.golden_tar)
 
         # Move kernel that comes with golden image.
-        extract_bootfiles(args, args.hostname == 'golden')
+        extract_bootfiles(args, args.is_golden)
```
[the remark this answers is not in the document]
Comment (coloroco) [reply utils/customize_node.py]: Sometimes it does make sense in testing environments or more generic "libraries".  I think just leaving it alone for the sake of closing this pull request makes sense, tool.

**Pull request merged**